bats/scripts/all-k3s.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+# Run BATS against a list of k3s versions.
+#
+# It will factory reset, so the current VM and settings will be lost.
+# Snapshots however should be preserved.
+
+set -o errexit -o nounset -o pipefail
+
+# VERSIONS is the list of versions to test, e.g. "1.28.6 1.29.1".
+# Special value "all" will be replaced with all available tags from GitHub.
+# Special value "latest" will be replaced with versions from the release channel.
+#
+# Downloading the airgap tarballs for all k3s releases take almost 50GB, so
+# make sure you have good bandwidth and enough storage. Maybe save the cache
+# for re-use before running BATS tests that will clear the cache.
+: "${VERSIONS:=}"
+
+# TEST can specify a filename with additional test commands. It will be sourced
+# after kubelet is responsive.
+: "${TEST:=}"
+
+# The VERSIONS list can be filtered by MIN and MAX versions. The MIN and MAX
+# versions themselves will not be filtered out.
+: "${MIN_VERSION:=1.0.0}"
+: "${MAX_VERSION:=1.999.0}"
+
+# When SKIP_RUN is non-empty then the tests will not be executed, but the script
+# filename will be written to stdout. Otherwise the script will be deleted.
+: "${SKIP_RUN:=}"
+
+# By default the BATS kubelet timeout is 10 minutes. If the test is expected to
+# fail starting up k3s a lot, then reducing the timeout it highly recommended
+# (there are almost 200 k3s versions).
+: "${RD_KUBELET_TIMEOUT:=2}"
+export RD_KUBELET_TIMEOUT
+
+if [[ $VERSIONS == "all" ]]; then
```

Comment:
It's odd-looking that `$VERSIONS` isn't quoted, given that there's an expectation for it to contain spaces. (But it works fine as-is.)